frontend/public/components/environment.jsx

```diff
@@ -185,14 +200,17 @@ export class EnvironmentPage extends PromiseComponent {
   }
 
   render() {
-    const {errorMessage, success, inProgress, currentEnvVars, stale} = this.state;
+    const {errorMessage, success, inProgress, currentEnvVars, stale, configMaps, secrets} = this.state;
     const {rawEnvData, readOnly, obj} = this.props;
 
+    if (!configMaps || !currentEnvVars) {
```

Comment:
Are secrets ever not available? @beanh66 and I were playing around with trying these edge cases and secrets seemed not be removable or at least regenerated. 

Thanks, the error handling above will help.

I think we want to block on loading the resources here because otherwise we would have to add spinners to all combo boxes.

Reply:
Yeah good catch on the loading state!
I forgot to add Secrets when adding the secret functionality. I'll add the check for secrets as well. 

I do think we should load the resources once right on mount. We could potentially have several combo boxes on that page that need the contents of secrets and config maps to be present in order to function with the current workflow.